Skeleton and conda from scratch tutorials (#3447)
conda/conda-build: docs/source/user-guide/tutorials/build-pkgs-skeleton.rst

   # Name                    Version                   Build  Channel
   ca-certificates           2019.1.23                     0 
   certifi                   2019.3.9                 py37_0 
   click                     7.0                      py37_0    local

Now verify that Click installed successfully:

.. code-block:: bash

    conda list

Scroll through the list until you find Click.

At this point you now have a conda package for Click that
can be installed in any conda environment of the same Python
version as your root environment. The remaining optional sections
show you how to make packages for other Python versions and other
architectures and how to upload them to your Anaconda.org account.

.. _`python-versions`:

Optional---Building for a different Python version
==================================================
By default, conda build creates packages for the version of
Python installed in the root environment. To build packages for
other versions of Python, you use the ``--python`` flag followed
by a version. For example, to explicitly build a version of the
Click package for Python 3, use:

.. code-block:: bash

>     conda-build --python 3 click

Reviewer: I think the reason why @soapy1 recommended python 2.7 is that it is different from what we show above.  You can supply any python version, even inexact ones like 2 or 3.  Unfortunately, this example doesn't show any capability here, because the outputs are the same as the packages above (all python 3.7)

Author: I changed it back to 2.7.

Notice that the file printed at the end of the ``conda-build``
output has changed to reflect the requested version of Python.
``conda install`` will look in the package directory for the file
that matches your current Python version.

Windows example file path:

.. code-block:: text

    C:\Users\jsmith\Miniconda\conda-bld\win-64\click-7.0-py37_0.tar.bz2

OS X example file path:

.. code-block:: text

    /Users/jsmith/miniconda/conda-bld/osx-64/click-7.0-py37_0.tar.bz2

Linux example file path:

.. code-block:: text

    /home/jsmith/miniconda/conda-bld/linux-64/click-7.0-py37_0.tar.bz2

.. note::
   Your path and filename will vary depending on your
   installation and operating system. Save the
   path and filename information for the next task.

.. _convert-conda-package:

Optional---Converting conda package for other platforms
========================================================

Now that you have built a package for your current platform with
conda build, you can convert it for use on other platforms with

> the ``conda convert`` command and a platform specifier from this

Reviewer: ```suggestion
the ``conda convert`` command.  This works only for pure python packages, where there is no compiled code.  Conda convert does nothing to change compiled code, it only adapts file paths to take advantage of the fact that python scripts are mostly platform-independent. Conda convert accepts a platform specifier from this
```

list:

* osx-64.
* linux-32.
* linux-64.
* win-32.
* win-64.
* all.

In the output directory, 1 folder will be created for each of the
1 or more platforms you chose and each folder will contain a
.tar.bz2 package file for that platform.

Windows:

.. code-block:: text

    conda convert -f --platform all C:\Users\jsmith\miniconda\conda-bld\win-64\click-7.0-py37_0.tar.bz2
    -o outputdir\

macOS and Linux:

.. code-block:: text

    conda convert --platform all /home/jsmith/miniconda/conda-bld/linux-64/click-7.0-py37_0.tar.bz2
    -o outputdir/

.. note::
   Change your path and filename to the exact path and
   filename you saved in :ref:`python-versions`.

To use these packages, you need to transfer them to other
computers and place them in the correct ``conda-bld/$ARCH``
directory for the platform, where ``$ARCH`` can be ``osx-64``,
``linux-32``, ``linux-64``, ``win-32``, or ``win-64``.

A simpler way is to upload all of the bz2 files to Anaconda.org
as described in the next task.

.. _`upload-to-anaconda-org`:

> 

Reviewer: ```suggestion
If you find yourself needing to use ``conda convert``, you might instead prefer to change your recipe to make your package a "noarch" package.  Noarch packages run anywhere, and do not require conda convert.  Some of the ecosystem tools don't yet support noarch packages, but for the most part, noarch packages are a better way to go.
```

.. code-block:: bash

    conda list

Scroll through the list until you find Click.

At this point you now have a conda package for Click that
can be installed in any conda environment of the same Python
version as your root environment. The remaining optional sections
show you how to make packages for other Python versions and other
architectures and how to upload them to your Anaconda.org account.

.. _`python-versions`:

Optional---Building for a different Python version
==================================================
By default, conda build creates packages for the version of
Python installed in the root environment. To build packages for
other versions of Python, you use the ``--python`` flag followed
by a version. For example, to explicitly build a version of the
Click package for Python 2.7, use:

.. code-block:: bash

    conda-build --python 2.7 click

Notice that the file printed at the end of the ``conda-build``
output has changed to reflect the requested version of Python.
``conda install`` will look in the package directory for the file
that matches your current Python version.

Windows example file path:

.. code-block:: text

>     C:\Users\jsmith\Miniconda\conda-bld\win-64\click-7.0-py37_0.tar.bz2

Reviewer: will be click-7.0-py27_0.tar.bz2

OS X example file path:

.. code-block:: text

>     /Users/jsmith/miniconda/conda-bld/osx-64/click-7.0-py37_0.tar.bz2

Reviewer: will be click-7.0-py27_0.tar.bz2

Linux example file path:

.. code-block:: text

>     /home/jsmith/miniconda/conda-bld/linux-64/click-7.0-py37_0.tar.bz2

Reviewer: ```suggestion
    /home/jsmith/miniconda/conda-bld/linux-64/click-7.0-py27_0.tar.bz2
```
